cv-data-controller/src/main/java/com/trihydro/cvdatacontroller/controller/CdotUpstreamPathController.java

```diff
@@ -27,366 +25,511 @@
+            JsonNode rootNode = objectMapper.readTree(routeJsonString);
+            rootNode.path("features").get(0).path("geometry").path("paths").get(0);
+        } catch (Exception e) {
+            return ResponseEntity.ok(false);
+        }
+        return ResponseEntity.ok(true);
     }
 
-    public List<Milepost> getAllMileposts() {
-      return allMileposts;
+    /**
+     * Retrieves all mileposts between the start and end points
+     *
+     * <p>This method uses `CdotGisService.RouteBetweenMeasures()` to retrieve the route between the start and end
+     * in JSON format. The JSON response contains all the latitude and longitude points in the path. This information
+     * is then extracted into the `Milepost` model and returned as a list of mileposts.</p>
+     *
+     * @param wydotTim The WydotTim object containing the data for the TIM.
+     * @return a list of `Milepost` objects representing the mileposts from the start to the end point of a Tim
+     * @throws JsonProcessingException if there is an error processing the JSON response
+     * @throws RestClientException     if an error occurs while making the request
+     */
+    @RequestMapping(method = RequestMethod.POST, produces = "application/json", value = "/get-milepost-start-end")
+    public ResponseEntity<List<Milepost>> getMilepostsByStartEndPoint(@RequestBody WydotTim wydotTim) throws JsonProcessingException,
+            RestClientException {
+        logger.warn("In MilePost correct version");
+        BigDecimal startLat = wydotTim.getStartPoint().getLatitude();
+        BigDecimal startLong = wydotTim.getStartPoint().getLongitude();
+        BigDecimal endLat = wydotTim.getEndPoint().getLatitude();
+        BigDecimal endLong = wydotTim.getEndPoint().getLongitude();
+        String routeId = wydotTim.getRoute().replace('-', '_');
+
+        ResponseEntity<String> startRouteDetails = cdotGisService.getMeasureAtPoint(startLong, startLat);
+        String startRouteJsonString = startRouteDetails.getBody();
+        ObjectMapper objectMapper = new ObjectMapper();
+        JsonNode rootNode = objectMapper.readTree(startRouteJsonString);
+        logger.info(rootNode.toString());
+        String startRoute = rootNode.path("features").get(0).path("attributes").get("Route").asText();
+        double startMeasure = rootNode.path("features").get(0).path("attributes").get("Measure").floatValue();
+
+        ResponseEntity<String> endRouteDetails = cdotGisService.getMeasureAtPoint(endLong, endLat);
+        String endRouteJsonString = endRouteDetails.getBody();
+        rootNode = objectMapper.readTree(endRouteJsonString);
+        String endRoute = rootNode.path("features").get(0).path("attributes").get("Route").asText();
+        double endMeasure = rootNode.path("features").get(0).path("attributes").get("Measure").floatValue();
+
+        if (!startRoute.equals(endRoute) || !startRoute.equals(routeId)) {
+            logger.warn("Unable to find route");
```

Comment:
Suggestion (non-blocking): Include some more information regarding why the conditional failed to resolve to true. Looks like there could be a few different cases here.

Reply:
Looks great!